src/tfrnnlm/tensorflow-rnnlm.h

```diff
@@ -0,0 +1,179 @@
+// tensorflow-rnnlm-lib.h
+
+// Copyright         2017 Hainan Xu
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef KALDI_TENSORFLOW_RNNLM_H_
+#define KALDI_TENSORFLOW_RNNLM_H_
+
+#include <string>
+#include <vector>
+#include "util/stl-utils.h"
+#include "base/kaldi-common.h"
+#include "fstext/deterministic-fst.h"
+#include "util/common-utils.h"
+#include "tensorflow/core/public/session.h"
+
+using tensorflow::Session;
+using tensorflow::Tensor;
+
+namespace kaldi {
+namespace tf_rnnlm {
+
+struct KaldiTfRnnlmWrapperOpts {
+  std::string unk_symbol;
+  int32 num_threads;  // 0 means unlimited
+
+  KaldiTfRnnlmWrapperOpts() : unk_symbol("<oos>"), num_threads(1) {}
+
+  void Register(OptionsItf *opts) {
+    opts->Register("unk-symbol", &unk_symbol, "Symbol for out-of-vocabulary "
+                   "words in rnnlm. (default = <oos>)");
+    opts->Register("num-threads", &num_threads, "Number of threads for TF computation; "
+                   "0 means unlimited. (default = 1)");
+  }
+};
+
+// this class wrapps the tensorflow based RNNLM and provides a set of interfaces
+// used in an OnDemandFST class used in lattice-rescoring
+//
+class KaldiTfRnnlmWrapper {
+ public:
+  // usually FSTs and the RNNLM would have different wordlists so we have 2
```

Comment:
Try to document this better, and use Doxygen-style documentation for most functions , documenting
their parameters.  This comment that you have here is more like a sequence of asides clarifications, it doesn't really clearly explain
from the top, what's going on, and doesn't make it easy to look up for a particular parameter, what it
is.  For inspiration, look at my PR about RNNLM stuff, and look for '@' which appears in Doxygen-style
comments.  Note: `///` and `/**` are used to start Doxygen comment blocks.